doc/api/esm.md

```diff
@@ -118,7 +118,7 @@ import url from 'url';
 export async function resolve(specifier, parentModuleURL, defaultResolver) {
   return {
     url: new URL(specifier, parentModuleURL).href,
-    format: 'esm'
+    translator: 'esm'
```

Comment:
Also this API for resolve was not intended to be extended beyond specifying the module format and extending to new module formats (I originally actually returned "loader" instead of "format", but from feedback in the last paragraph at https://github.com/nodejs/node/pull/15445#issuecomment-330259441 restricted this).

What's also nice about resolve returning just "format" is one can think of the resolve hook as being equally applicable for browser resolution (if/when a browser resolver ever gets implemented).  I know it's somewhat of a dream that these might match up in future, but I'd still like to try and shoot for it.

Reply:
would the path moving forward here to basically `s/translator/format/` for all the properties and file names, back to their originals?

Reply:
I'm not sure, I was just trying to share the reasoning behind using the term "format". If you're asking my preference, yes I do think the name should stay, but I'm more than happy to discuss the terminology further.